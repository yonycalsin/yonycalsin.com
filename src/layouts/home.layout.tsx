import * as React from 'react'

import { FloatSocials, Navbar } from '~/components'
import NightModeButton from '~/components/night-mode-button'

type HomeLayoutProps = {
  children: React.ReactNode
  isStandaloneBrand?: boolean
}

export const HomeLayout = (props: HomeLayoutProps) => {
  const { children } = props

  return (
    <>
      <NightModeButton />
      <Navbar />

      <div className="container">{children}</div>

      <footer className="bg-gray-50 dark:bg-gray-900">
        <div className="container border-t-2 border-gray-100 dark:border-gray-700">
          <div className="flex flex-col md:flex-row flex-wrap justify-between items-center ">
            <p>
              <small>
                Created By{' '}
                <i>
                  <b>Yony Calsin</b>
                </i>{' '}
              </small>
            </p>

            <p>
              <small>With NextJs, Typescript, TailwindCss, and Vercel</small>
            </p>
          </div>
        </div>
      </footer>

      <FloatSocials />
    </>
  )
}
