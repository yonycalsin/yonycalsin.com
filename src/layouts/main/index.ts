export * from './components'

// @yonycalsin/eslint-import-sort/typescript-react-exports group
export { default as MainLayout } from './main'
