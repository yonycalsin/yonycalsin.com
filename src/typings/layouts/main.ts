import type * as React from 'react'

interface MainLayoutProps {
  children: React.ReactNode
}

export type { MainLayoutProps }
