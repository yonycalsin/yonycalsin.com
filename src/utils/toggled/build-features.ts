import type { DefaultFeature } from 'toggled'

import type { Features } from 'utils/constants'

function buildFeatures(features: Partial<Record<Features, boolean>>) {
  const buildFeatures: DefaultFeature[] = []

  for (const [feature, value] of Object.entries(features)) {
    if (!value) {
      continue
    }

    buildFeatures.push({
      slug: feature,
    })
  }

  return buildFeatures
}

export default buildFeatures
