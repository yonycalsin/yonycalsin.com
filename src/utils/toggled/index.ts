export { default as buildFeatures } from './build-features'
