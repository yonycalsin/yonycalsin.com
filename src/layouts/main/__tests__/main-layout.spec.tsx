import { ChakraKBar } from '@chakra-ui-kbar/core'
import { render } from '@testing-library/react'
import { FeatureProvider } from 'toggled'

import { MainLayout } from 'layouts'

describe('MainLayout', () => {
  it('renders the main layout', () => {
    const view = render(
      <FeatureProvider features={[]}>
        <ChakraKBar>
          <MainLayout>
            <h1>Content</h1>
          </MainLayout>
        </ChakraKBar>
      </FeatureProvider>,
    )

    expect(view.baseElement).toMatchSnapshot()
  })
})
