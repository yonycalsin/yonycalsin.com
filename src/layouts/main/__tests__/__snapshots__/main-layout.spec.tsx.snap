// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders the main layout 1`] = `
.emotion-0 {
  width: full;
  top: 0px;
  z-index: banner;
  left: 0px;
  box-shadow: xs;
}

.chakra-ui-dark .emotion-0:not([data-theme]),
[data-theme=dark] .emotion-0:not([data-theme]),
.emotion-0[data-theme=dark] {
  box-shadow: lg;
}

.emotion-1 {
  max-width: container.md;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 py {
  base: 5;
  md: 8;
}

.emotion-2 {
  display: contents;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 4px;
  margin-inline-start: 4px;
}

.emotion-4 {
  color: primary.500;
  font-weight: bold;
}

.emotion-6 {
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-7 {
  max-width: container.md;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-9 {
  font-weight: bold;
}

.emotion-10 {
  color: gray.600;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-12 {
  font-family: mono;
  -webkit-margin-start: 2px;
  margin-inline-start: 2px;
  -webkit-margin-end: 2px;
  margin-inline-end: 2px;
}

.emotion-14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  bottom: 8px;
  left: 8px;
}

.emotion-14>*:not(style)~*:not(style) {
  margin-top: 5px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-14 display {
  base: none;
  lg: flex;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        class="chakra-container emotion-1"
      >
        <a
          class="chakra-link emotion-2"
          href="/"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt=""
              data-nimg="intrinsic"
              decoding="async"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </a>
        <div
          class="chakra-stack emotion-3"
        >
          <a
            class="chakra-link emotion-4"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="chakra-link emotion-4"
            href="/me"
          >
            About Me
          </a>
        </div>
      </div>
    </div>
    <div
      style="opacity: 0; transform: translateY(8px) translateZ(0);"
    >
      <h1>
        Content
      </h1>
    </div>
    <footer
      class="emotion-6"
    >
      <div
        class="chakra-container emotion-7"
      >
        <div
          class="chakra-stack emotion-8"
        >
          <p
            class="chakra-text emotion-9"
          >
            Created By Yony Calsin
          </p>
          <p
            class="chakra-text emotion-10"
          >
            Built with Next.js, MDX, Chakra-Ui and Vercel
          </p>
          <button
            class="chakra-button emotion-11"
            type="button"
          >
            Press 
            <kbd
              class="chakra-kbd emotion-12"
            >
              Ctrl
            </kbd>
            +
            <kbd
              class="chakra-kbd emotion-12"
            >
              K
            </kbd>
             to start â†’
          </button>
        </div>
      </div>
    </footer>
    <div
      class="chakra-stack emotion-14"
    >
      <a
        class="chakra-link emotion-15"
        href="https://github.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
      <a
        class="chakra-link emotion-15"
        href="https://www.linkedin.com/in/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Linkedin
      </a>
      <a
        class="chakra-link emotion-15"
        href="https://twitter.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        class="chakra-link emotion-15"
        href="mailto:helloyonycalsin@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        Email
      </a>
    </div>
  </div>
</body>
`;
